rasa/core/channels/console.py

```diff
@@ -137,14 +146,20 @@ def get_user_input(button_question: questionary.Question) -> Optional[Text]:
             bot_responses = send_message_receive_stream(
                 server_url, auth_token, sender_id, text
             )
+            previous_response = None
```

Comment:
Functionality inside "if" and "else" looks pretty similar, can we extract common parts and put them after if-else?

Reply:
I would love to, but the "pretty similar" code is inside of for loops, from which one is async... Meaning I cannot extract it into a simple `for` loop or `async for` loop. In case there is some python way to achieve this, I'll gladly rewrite it.

Reply:
Yes, you're right. My bad. I thought that there's async generator in both of the loops.
I guess that could be refactored to look the same, but it's definitely out of scope for this PR.